Documentation for Submodules

jit-1 opened the pull request:
# Description

Documentation for Submodules using Autodocs in Sphinx

## Type of change

Please delete options that are not relevant.

- [x] This change requires a documentation update


# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

ChristinaLast committed:
refactor: merging changes from develop

ChristinaLast committed:
refactor: merging develop

ChristinaLast committed:
refactor: merging code corresponding to issue 17

ChristinaLast committed:
style: adding arguements to format statement

ChristinaLast committed:
refactor: removing unused imports

ChristinaLast committed:
refactor: pulling develop

jit-1 committed:
docs: restructuring docs

jit-1 committed:
docs:index.txt initialized

jit-1 committed:
docs: index modified

jit-1 committed:
docs: index modified

jit-1 committed:
docs: Restore index.bak with original content

jit-1 committed:
docs: Add contect index.rst

jit-1 committed:
docs: Add content index.rst

jit-1 committed:
Merge branch '47-completing-open-source-implementation' of https://github.com/AQ-AI/openaq-engine into 47-completing-open-source-implementation

jit-1 committed:
docs: Restore docker_setup.html

jit-1 committed:
docs: Update index.rst

ChristinaLast committed:
feat: adding new sphinx build with additional documentation

ChristinaLast committed:
refactor: removing unneeded file

ChristinaLast committed:
47 completing open source implementation (#48)

ChristinaLast committed:
refactor: merging develop

ChristinaLast committed:
refactor: adding ignore='W503' to pre-commit configuration

ChristinaLast committed:
style: sorting imports

ChristinaLast committed:
refactor: merging develop

ChristinaLast committed:
refactor: Merge branch '39-add-weighted-average-to-satellite-data-extraction-to-reduce-nulls' into 17-adding-satellite-data-features

ChristinaLast committed:
docs: adding reminder to export environmment vairables

ChristinaLast committed:
refactor: linting

ChristinaLast committed:
refactor: adding configurations needed to EEConfig and adjusting docstring in cohort-builder

ChristinaLast committed:
refactor: adjusting categorical feature generation configuration

ChristinaLast committed:
refactor: removing print statement

ChristinaLast committed:
refactor: removing df.pipe() to ensure correct write to dbase

ChristinaLast committed:
17 adding satellite data features (#24)

ChristinaLast committed:
refactor: adding window_no

Commit:
refactor: adjusting package versions

ChristinaLast committed:
feat: removing openssl from git tracking

ChristinaLast committed:
feat: adding mlflow experiment run details

ChristinaLast committed:
refactor: removing alias to stop flake8 error from triggering

ChristinaLast committed:
feat: adding additional env variables

ChristinaLast committed:
refactor: adding openssl to gitignore

ChristinaLast committed:
refactor: defining imports

ChristinaLast committed:
style: linting

ChristinaLast committed:
49 initial mlflow test (#50)

ChristinaLast committed:
refactor: removing unused assignments for linting

ChristinaLast committed:
feat: updating packages

ChristinaLast committed:
style: removing unused imports

ChristinaLast committed:
51 model training run (#52)

ChristinaLast committed:
refactor: adding stashed changes

ChristinaLast committed:
refactor: merging develop

ChristinaLast committed:
refactor: adjusting arguements for API call

ChristinaLast committed:
refactor: adding WIP evaluator

ChristinaLast committed:
refactor: adding evaluation scripts

ChristinaLast committed:
style: flake8 linting

ChristinaLast committed:
refactor: removing joblib files

ChristinaLast committed:
refactor: removing models from dir

ChristinaLast committed:
feat: writing to results table

ChristinaLast committed:
style: sorting imports

ChristinaLast committed:
style: removing printing

ChristinaLast committed:
refactor: adjusting amount of results returned cohorts

ChristinaLast committed:
refactor: on gitignore

jit-1 committed:
refactor: Adding gcloud to Poetry, mlruns added to gitignore

ChristinaLast committed:
feat: adding country-specific filtering (WIP)

jit-1 committed:
Update CODE_OF_CONDUCT.md

jit-1 committed:
Update CODE_OF_CONDUCT.md

jit-1 committed:
Update CODE_OF_CONDUCT.md

ChristinaLast committed:
feat: adding adjustments to pipeline

ChristinaLast committed:
fix: adding end of file fixer

ChristinaLast committed:
fix: removing bad pattern in .gitignore

ChristinaLast committed:
Merge branch 'master' of https://github.com/AQ-AI/openaq-engine into 50-creating-query-parameters-for-intra-city-run

ChristinaLast committed:
fix: adding pytest-mock and upgrading mlflow

ChristinaLast committed:
test: adding initial unittests

ChristinaLast committed:
refactor: adding coverage as dev dependency

ChristinaLast committed:
feat: adding build features as a cli command

ChristinaLast committed:
style: removing print statements

ChristinaLast committed:
feat: adding tests for time_splitter

ChristinaLast committed:
fix: correcting typo

ChristinaLast committed:
fix: removing unused tests

ChristinaLast committed:
feat: adding cohort_builder tests

ChristinaLast committed:
style: removing print statements

ChristinaLast committed:
style: removing print statements

ChristinaLast committed:
feat: adding cohort builder tests

ChristinaLast committed:
adding WIP tests

ChristinaLast committed:
Merge branch 'develop' of https://github.com/AQ-AI/openaq-engine into 58-separate-testing-and-evaluation-code

ChristinaLast committed:
58 separate testing and evaluation code (#59)

ChristinaLast committed:
fix: adding updates to pyyaml

ChristinaLast committed:
adding docstrings in cohort builder

ChristinaLast committed:
fix: adjusting parameters to successfully return timesplits

ChristinaLast committed:
fix: adding correct columns for feature builder

ChristinaLast committed:
fix: linting and auto sorting imports

ChristinaLast committed:
fix: removing get_data function

ChristinaLast committed:
fix: linting and sorting imports

ChristinaLast committed:
fix: adding local_data argument to  linting and formatting:

ChristinaLast committed:
fix: linting and sorting

ChristinaLast committed:
fix: removing unused imports

ChristinaLast committed:
feat: updating pyproj.toml to 3.9

ChristinaLast committed:
fix: removing unused engine and imports

ChristinaLast committed:
fix: add newline

ChristinaLast committed:
fix: adding whitespace

ChristinaLast committed:
fix: linting

ChristinaLast committed:
fix: reformatting

ChristinaLast committed:
fix: linting

ChristinaLast committed:
style: linting

ChristinaLast committed:
style: isort and black issues

ChristinaLast committed:
feat: updating pre-commit configuration

ChristinaLast committed:
feat: rebuilding pypyorject.toml

ChristinaLast committed:
fix: removing poetry.lock file from git branch

ChristinaLast committed:
fix: cohort_builder

ChristinaLast committed:
fix: adding updated pyproject.toml

ChristinaLast committed:
tests: adding test suite

ChristinaLast committed:
Update workflow.yaml

ChristinaLast committed:
fix: adding config to understand relative imports for testing

ChristinaLast committed:
fix: Merge branch '52-creating-query-parameters-for-intra-city-run' of https://github.com/AQ-AI/openaq-engine into 53-creating-query-parameters-for-intra-city-run

ChristinaLast committed:
fix: adding src

ChristinaLast committed:
fix: tests

ChristinaLast committed:
fix: adjusting relative imports

ChristinaLast committed:
fix: removing openaq_engine from relative imports

ChristinaLast committed:
fix: linting

ChristinaLast committed:
feat: adding editable installation

ChristinaLast committed:
feat: adding python path

ChristinaLast committed:
feat: editting local install

ChristinaLast committed:
fix: adding poetry shell

ChristinaLast committed:
fix: removing 3.8 from workflow:

ChristinaLast committed:
fix: editting to remove poetry shell

ChristinaLast committed:
fix: linting .env

ChristinaLast committed:
fix: adding run: to workflow.yaml

ChristinaLast committed:
fix: removing openaq_engine from local imports

ChristinaLast committed:
fix: adding sqlalchemy import in correct location

ChristinaLast committed:
fix: adding test_db params to workflow

ChristinaLast committed:
fix: changing version

ChristinaLast committed:
fix: reformatting

ChristinaLast committed:
fix: removing imports

ChristinaLast committed:
feat: adding env to time_splitter tests

ChristinaLast committed:
fix: removing imports

ChristinaLast committed:
fix: imports

ChristinaLast committed:
fix: linting

ChristinaLast committed:
fix: utils

ChristinaLast committed:
fix: merging develop?

ChristinaLast committed:
updating poetry lock file

ChristinaLast committed:
fix: re-naming cohorts_db

ChristinaLast committed:
fix: merge conflicts

ChristinaLast committed:
feat: adding to gitignore

ChristinaLast committed:
feat: Merge branch '53-creating-query-parameters-for-intra-city-run' of github.com:AQ-AI/openaq-engine into 64-mongolia-local-model-steps-to-build

ChristinaLast committed:
fix: adjusting name to process MN cohorts with supplemented local data

jit-1 committed:
53 creating query parameters for intra city run (#54)

ChristinaLast committed:
Revert "53 creating query parameters for intra city run"

ChristinaLast committed:
fix: removing bespoke settings

ChristinaLast committed:
fix: fixing tests

ChristinaLast committed:
Revert "53 creating query parameters for intra city run" (#67)

ChristinaLast committed:
fix: tests

ChristinaLast committed:
fix: adding script to process satellite

ChristinaLast committed:
fix: adding script to process satellite

jit-1 committed:
openaq-engine readme update

jit-1 committed:
Create CNAME

jit-1 committed:
Create CNAME

jit-1 committed:
Delete CNAME

jit-1 committed:
Delete CNAME

jit-1 committed:
Create CNAME

jit-1 committed:
Delete CNAME

ChristinaLast committed:
fix: removing multiple definitions

ChristinaLast committed:
fix: linting

ChristinaLast committed:
feat: restoring tests

ChristinaLast committed:
fix: linting

ChristinaLast committed:
fix: removing unused library

ChristinaLast committed:
fix: updating to add pytest-mock

ChristinaLast committed:
fix: removing unused imports

ChristinaLast committed:
Fix: linting

ChristinaLast committed:
fix: adjusting test

ChristinaLast committed:
fix: linting

ChristinaLast committed:
fix: linting and adding missing import

ChristinaLast committed:
fix: linting

ChristinaLast committed:
fix: linting

ChristinaLast committed:
fix: adjusting test

ChristinaLast committed:
fix: removing unused import

ChristinaLast committed:
fix: importing datetime

ChristinaLast committed:
fix: linting

ChristinaLast committed:
fix: linting

ChristinaLast committed:
fix: adjusting to ensure relative datetime imports work

ChristinaLast committed:
fix: adding print

ChristinaLast committed:
fix: linting

ChristinaLast committed:
53 creating query parameters for intra city run (#68)

ChristinaLast committed:
feat: adding model Card documentation

ChristinaLast committed:
feat: adding model Card documentation (#71)

ChristinaLast committed:
docs: moving graphana build docs

ChristinaLast committed:
docs: adding mlflow instructions

ChristinaLast committed:
docs: adding data schema for model generation

ChristinaLast committed:
docs: adding software architecture

jit-1 committed:
docs: Add readme for setuping up locally

jit-1 committed:
docs: Add readme for setuping up locally

ChristinaLast committed:
72 documentation in a githubio for example explaining how the data extraction and processing workflows (#73)

ChristinaLast committed:
Merge branch 'develop' into 67-documentation-local

ChristinaLast commented on docs/source/openaq_engine.src.preprocessing.rst:
I am checking the submodule documentation - which branch are you doing it from? it seems that most of the functions are missing.

ChristinaLast committed:
fix: adding poetry.lock

ChristinaLast committed:
style: removing unused import

ChristinaLast committed:
fix: removing unneeded folders from .git history

jit-1 commented:
Documentation to look at: 

1. Autodoc Config Section: [https://sphinx-rtd-tutorial.readthedocs.io/en/latest/sphinx-config.html](https://sphinx-rtd-tutorial.readthedocs.io/en/latest/sphinx-config.html)
2. How the docstrings need to be written in the Modules:  [https://sphinx-rtd-tutorial.readthedocs.io/en/latest/docstrings.html](https://sphinx-rtd-tutorial.readthedocs.io/en/latest/docstrings.html)

ChristinaLast committed:
fix: adding .gitignore folders

ChristinaLast committed:
Remove mlruns directory from version control

ChristinaLast committed:
fix: linting

ChristinaLast committed:
fix: updating yaml

ChristinaLast committed:
fix: updating build-system

ChristinaLast committed:
fix: updating poetry env

ChristinaLast committed:
fix: env

jit-1 committed:
Update sphinx.yaml

Numpy Error resolution: --no-use-pep517 error: It is not possible to use --no-use-pep517 without setuptools and wheel installed.

jit-1 committed:
Update sphinx.yaml

The process of installing numpy==1.24.4 failed due to several issues related to the Python environment and the build process.
> pip install cython
> 3.x to 3.9

jit-1 committed:
Update sphinx.yaml

Avoid 3.12 for the numpy error
for pip install numpy
Remove ----no-use-pep517 
Using --prefer-binary

ChristinaLast committed:
fix: simplifying pyproject.toml

ChristinaLast committed:
fix: simplifying sphinx

ChristinaLast committed:
fix: updating to python 3.12

ChristinaLast committed:
fix: updating poetry.lock to match pyproject

ChristinaLast committed:
Merge branch 'develop' of https://github.com/AQ-AI/openaq-engine into 67-documentation-local

ChristinaLast committed:
fix: updating poetry env

ChristinaLast committed:
fix: updating pyproject.toml

ChristinaLast committed:
fix: poetry env

ChristinaLast committed:
fix: adding mlflow

ChristinaLast committed:
fix: adding setuptools

ChristinaLast committed:
fix: updating numpy version

ChristinaLast committed:
fix: updating poetry

ChristinaLast committed:
linting

ChristinaLast committed:
fix: adding new libraries

ChristinaLast committed:
feat: updating conf.py for sphinx documentation

ChristinaLast committed:
fix: linting test

ChristinaLast committed:
fix: linting tests

ChristinaLast committed:
fix: updating test

ChristinaLast committed:
fix: adding parameter column to test data

ChristinaLast committed:
fix: adjusting sample data

ChristinaLast committed:
fix: adjusting sample data

ChristinaLast committed:
Fix: linting

ChristinaLast committed:
fix: adjusting config to filter countries

ChristinaLast committed:
fix: linting

ChristinaLast committed:
fix: adding tying

ChristinaLast committed:
fix: linting

ChristinaLast committed:
fix: linting

ChristinaLast committed:
fix: adding city column

ChristinaLast committed:
fix: adding city

ChristinaLast committed:
fix: linting

ChristinaLast committed:
fix: linting

ChristinaLast committed:
fix: linting

ChristinaLast committed:
fix: linting

ChristinaLast committed:
fix: adding debug statements

ChristinaLast committed:
fix: adding test countries

ChristinaLast committed:
fix: adjusting boolean logic in module

ChristinaLast committed:
fix: print filter default

ChristinaLast committed:
fix: adding additional filters

ChristinaLast committed:
fix: linting

ChristinaLast committed:
fix: lintin

ChristinaLast committed:
fix: linting

ChristinaLast committed:
fix: linting

ChristinaLast committed:
fix: linting

ChristinaLast committed:
fix: linting

ChristinaLast committed:
fix: removing assert statements

ChristinaLast committed:
fix: linting

ChristinaLast committed:
fix: adding mock

ChristinaLast committed:
fix: adjusting test

ChristinaLast committed:
fix: adjusting test

ChristinaLast reviewed (commented):
Completed the docs and testing
